action.yaml

```diff
@@ -130,6 +130,45 @@ runs:
       env:
         GITHUB_ACTION_PATH: ${{ github.action_path }}
 
+    - name: Set Trivy environment variables
+      shell: bash
+      run: |
+        # Note: There is currently no way to distinguish between undefined variables and empty strings in GitHub Actions.
+        # This limitation affects how we handle default values and empty inputs.
+        # For more information, see: https://github.com/actions/runner/issues/924
+
+        # Function to set environment variable only if the input is provided and different from default
+        set_env_var_if_provided() {
+          local var_name="$1"
+          local input_value="$2"
+          local default_value="$3"
+        
+          if [ ! -z "$input_value" ] && [ "$input_value" != "$default_value" ]; then
+            echo "$var_name=$input_value" >> $GITHUB_ENV
+          fi
+        }
+        
+        # Set environment variables, handling those with default values
+        # cf. https://aquasecurity.github.io/trivy/latest/docs/configuration/#environment-variables
+        set_env_var_if_provided "TRIVY_INPUT" "${{ inputs.input }}" ""
+        set_env_var_if_provided "TRIVY_EXIT_CODE" "${{ inputs.exit-code }}" ""
```

Comment:
If it isn't set, is the default equivalent to a zero for this case?

Reply:
I'm aware that would be the default Trivy behavior (if exit code is not specified), I am just not sure how GitHub Actions interprets an empty string in this case.

Reply:
>If it isn't set, is the default equivalent to a zero for this case?

The action will not set `TRIVY_EXIT_CODE` and Trivy CLI just uses the default value.

Reply:
>I am just not sure how GitHub Actions interprets an empty string in this case.

If [the input is empty or the same as a default value](https://github.com/aquasecurity/trivy-action/blob/0a8669cd411537772694157727f914fcffeff683/action.yaml#L146-L148), it doesn't set an environment variable.